<h1 align=center><font size = 4>Capstone Project- Battle of Neighborhoods </font></h1>

## Opening a Pizza Place in Suburbs of Chicago

### Introduction
The primary objective of this project is to locate the ideal area to start a Pizza Place in one of the many suburbs of Chicago. Chicago is the most populous city in Illinois, and the third most populous city in the United States. Chicago is an international hub for finance, culture, commerce, industry, education, technology, telecommunications, and transportation. Since it's very expensive to live in the city, many people working in the city choose to live in the suburbs of Chicago. Some of the nearby suburbs such as Aurora, Naperville, Joliet etc have transits to Chicago so that people can get to the city for work. These places have phenomenal malls, supermarkets, restaurants etc.
Chicago already has lots of pizza places and the cost of setting up a new restaurant in Chicago will be more expensive compared to its nearby cities. Rather than opening a Pizza place in Chicago, one can look into opening one in the nearby expanding cities. First, we have to find the nearby cities and number of pizza places already in these areas. Then we have to look into how populated these cities are. Opening a restaurant in a place that is less populated may not bring much revenue, whereas opening one in a populated area will bring more competition to business.

### The Target Audience
The target audience for this project is any business owner interested in opening a Pizza Place in the Chicagoland area. In fact by making small changes on what venue data is being analyzed, the same principle can be used to find an ideal location for any business. Starting a new business in the Chicagoland area will be more optimal than starting one in the main city. For example the rent for a building in Chicago will be more compared to its nearby cities.

### Data
In order to find the ideal location to open a new Pizza Place we will need the following data,

• List of the neighboring cities of Chicago with the population.
The above data is obtained from Wikipedia page https://en.wikipedia.org/wiki/Chicago_metropolitan_area. This page gives the nearby cities of Chicago with their population. Using data scrapping we get the city names and the population of each city. This will give us an idea about how thickly or thinly are these places populated. Below is the data that we get from the wikipedia website.

• Venue Data of the above cities
We will use Foursquare API for getting this information. By uing the API we can get all the venues for the above cities and conduct an analysis on the Pizza Places in each of the cities. Below is the sample data that is optained using Foursquare API

First let's install all the packages and import the libraries

In [1]:
!pip install folium
!pip install geocoder

     |████████████████████████████████| 102kB 5.3MB/s ta 0:00:01


In [2]:
!pip install beautifulsoup4
!pip install geopy

     |████████████████████████████████| 122kB 4.7MB/s eta 0:00:01
     |████████████████████████████████| 112kB 6.1MB/s eta 0:00:01


In [3]:
import matplotlib.cm as cm
import matplotlib.colors as colors

import folium # map rendering library
from bs4 import BeautifulSoup

import requests

from geopy.geocoders import Nominatim
import pandas as pd
import numpy as np


### Methodology

We will get the names and population of nearby cities of Chicago

In [4]:
Chicago_page = requests.get("https://en.wikipedia.org/wiki/Chicago_metropolitan_area")
Chicago_page

<Response [200]>

Using Beautifulsoup we will extract the required information from the wikipedia website

In [5]:
# Using Beautifulsoup we will get the required information from the wikipedia website
soup = BeautifulSoup(Chicago_page.content, 'html.parser')
section = soup.find(id="Over_100,000_population").parent
Neighboring_sub = section.find_next('ul').find_all('li')
records=[]
suburb_population=[]
for elem in Neighboring_sub:
    records.append(elem.text.strip().split('(')[0])
    suburb_population.append(elem.text)
records    


['Aurora, Illinois ',
 'Naperville, Illinois ',
 'Joliet, Illinois ',
 'Elgin, Illinois ',
 'Kenosha, Wisconsin ']

In [6]:
section = soup.find(id="Over_50,000_population").parent
Neighboring_sub = section.find_next('ul').find_all('li')
for elem in Neighboring_sub:
    records.append(elem.text.strip().split('(')[0])
    suburb_population.append(elem.text)
records 

['Aurora, Illinois ',
 'Naperville, Illinois ',
 'Joliet, Illinois ',
 'Elgin, Illinois ',
 'Kenosha, Wisconsin ',
 'Waukegan, Illinois ',
 'Cicero, Illinois ',
 'Bolingbrook, Illinois ',
 'Arlington Heights, Illinois ',
 'Hammond, Indiana ',
 'Gary, Indiana ',
 'Evanston, Illinois ',
 'Schaumburg, Illinois ',
 'Palatine, Illinois ',
 'Skokie, Illinois ',
 'Des Plaines, Illinois ',
 'Orland Park, Illinois ',
 'Tinley Park, Illinois ',
 'Oak Lawn, Illinois',
 'Berwyn, Illinois ',
 'Mount Prospect, Illinois ',
 'Wheaton, Illinois ',
 'Oak Park, Illinois ']

We extracted the required information. Now we will have to clean and format the data so that we have a data frame with the nearby cities and its population

In [7]:
suburb_population
column_names1 = ['suburb', 'population']

# instantiate the dataframe
suburb_population_format = pd.DataFrame(columns=column_names1)
for i,pop in enumerate(suburb_population):
    suburb=pop.split('(')[0]
    if suburb != "Oak Lawn, Illinois":
        pop1=int(pop[pop.find('(')+len("("):pop.rfind(")")].replace(',',''))
        suburb_population_format = suburb_population_format.append({'suburb': suburb,'population': pop1}, ignore_index=True)
        
suburb_population_format    

suburb population
0              Aurora, Illinois      198870
1          Naperville, Illinois      149196
2              Joliet, Illinois      148227
3               Elgin, Illinois      111401
4            Kenosha, Wisconsin      101124
5            Waukegan, Illinois       85720
6              Cicero, Illinois       79943
7         Bolingbrook, Illinois       76468
8   Arlington Heights, Illinois       74593
9              Hammond, Indiana       74423
10                Gary, Indiana       74879
11           Evanston, Illinois       72836
12         Schaumburg, Illinois       72387
13           Palatine, Illinois       67143
14             Skokie, Illinois       62088
15        Des Plaines, Illinois       58243
16        Orland Park, Illinois       57634
17        Tinley Park, Illinois       55446
18             Berwyn, Illinois       53803
19     Mount Prospect, Illinois       53218
20            Wheaton, Illinois       52914
21           Oak Park, Illinois       52443

Next we will get the Latitude and Longitude information of all the above cities

In [8]:
column_names = ['Suburb', 'Latitude', 'Longitude'] 

# instantiate the dataframe
NearBySuburb = pd.DataFrame(columns=column_names)

for elem in records:
    suburb_name=elem
    geolocator = Nominatim(user_agent="chi_explorer")
    location = geolocator.geocode(elem)
    sub_lat=location.latitude
    sub_lon=location.longitude
    NearBySuburb = NearBySuburb.append({'Suburb': suburb_name,
                                          'Latitude': sub_lat,
                                          'Longitude': sub_lon}, ignore_index=True)
NearBySuburb  

Suburb   Latitude  Longitude
0              Aurora, Illinois   41.757170 -88.314754
1          Naperville, Illinois   41.772870 -88.147928
2              Joliet, Illinois   41.526360 -88.084021
3               Elgin, Illinois   42.037260 -88.281099
4            Kenosha, Wisconsin   42.584677 -87.821226
5            Waukegan, Illinois   42.363633 -87.844794
6              Cicero, Illinois   41.845540 -87.754020
7         Bolingbrook, Illinois   41.700330 -88.071771
8   Arlington Heights, Illinois   42.081156 -87.980216
9              Hammond, Indiana   41.583366 -87.500043
10                Gary, Indiana   41.602129 -87.337137
11           Evanston, Illinois   42.044739 -87.693046
12         Schaumburg, Illinois   42.033361 -88.083406
13           Palatine, Illinois   42.110578 -88.033689
14             Skokie, Illinois   42.033369 -87.733397
15        Des Plaines, Illinois   42.041582 -87.887392
16        Orland Park, Illinois   41.630663 -87.853629
17        Tinley Park, Illinois   41.573367 -87.784494
18            Oak Lawn, Illinois  41.710866 -87.758108
19             Berwyn, Illinois   41.850587 -87.793668
20     Mount Prospect, Illinois   42.066417 -87.937291
21            Wheaton, Illinois   41.864696 -88.110171
22           Oak Park, Illinois   41.887814 -87.788762

Next using Foursquare API we will explore all the places in the above cities

In [9]:
CLIENT_ID = 'LEUWLT3BHS0W4FKXSFL3GOGCBVHMZDY0M1IXTLBGS44DOUCG' # your Foursquare ID
CLIENT_SECRET = 'CG3HWM5SPRWUDUEORPNUGU3AH4SU03RWD2LHGM5A4FQYB0KW' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version
LIMIT = 100 # A default Foursquare API limit value
print('Your credentails:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Your credentails:
CLIENT_ID: LEUWLT3BHS0W4FKXSFL3GOGCBVHMZDY0M1IXTLBGS44DOUCG
CLIENT_SECRET:CG3HWM5SPRWUDUEORPNUGU3AH4SU03RWD2LHGM5A4FQYB0KW


In [10]:
def getNearbyVenues(names, latitudes, longitudes, radius=1500):
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Suburb', 
                  'Suburb Latitude', 
                  'Suburb Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [11]:
ChicagoSuburbVenues = getNearbyVenues(names=NearBySuburb['Suburb'],
                                   latitudes=NearBySuburb['Latitude'],
                                   longitudes=NearBySuburb['Longitude']
                                  )

Aurora, Illinois 
Naperville, Illinois 
Joliet, Illinois 
Elgin, Illinois 
Kenosha, Wisconsin 
Waukegan, Illinois 
Cicero, Illinois 
Bolingbrook, Illinois 
Arlington Heights, Illinois 
Hammond, Indiana 
Gary, Indiana 
Evanston, Illinois 
Schaumburg, Illinois 
Palatine, Illinois 
Skokie, Illinois 
Des Plaines, Illinois 
Orland Park, Illinois 
Tinley Park, Illinois 
Oak Lawn, Illinois
Berwyn, Illinois 
Mount Prospect, Illinois 
Wheaton, Illinois 
Oak Park, Illinois 


In [12]:
ChicagoSuburbVenues.shape

(1377, 7)

From the above value now we know 1377 venues were retrieved
Now let us look at sample of the data.

In [13]:
ChicagoSuburbVenues.head(25)

Suburb  Suburb Latitude  Suburb Longitude  \
0   Aurora, Illinois          41.75717        -88.314754   
1   Aurora, Illinois          41.75717        -88.314754   
2   Aurora, Illinois          41.75717        -88.314754   
3   Aurora, Illinois          41.75717        -88.314754   
4   Aurora, Illinois          41.75717        -88.314754   
5   Aurora, Illinois          41.75717        -88.314754   
6   Aurora, Illinois          41.75717        -88.314754   
7   Aurora, Illinois          41.75717        -88.314754   
8   Aurora, Illinois          41.75717        -88.314754   
9   Aurora, Illinois          41.75717        -88.314754   
10  Aurora, Illinois          41.75717        -88.314754   
11  Aurora, Illinois          41.75717        -88.314754   
12  Aurora, Illinois          41.75717        -88.314754   
13  Aurora, Illinois          41.75717        -88.314754   
14  Aurora, Illinois          41.75717        -88.314754   
15  Aurora, Illinois          41.75717        -88.314754   
16  Aurora, Illinois          41.75717        -88.314754   
17  Aurora, Illinois          41.75717        -88.314754   
18  Aurora, Illinois          41.75717        -88.314754   
19  Aurora, Illinois          41.75717        -88.314754   
20  Aurora, Illinois          41.75717        -88.314754   
21  Aurora, Illinois          41.75717        -88.314754   
22  Aurora, Illinois          41.75717        -88.314754   
23  Aurora, Illinois          41.75717        -88.314754   
24  Aurora, Illinois          41.75717        -88.314754   

                                  Venue  Venue Latitude  Venue Longitude  \
0                     Paramount Theatre       41.757414       -88.314938   
1                           Gillerson's       41.759606       -88.315031   
2                         Endiro Coffee       41.759559       -88.314812   
3                 Tecalitlan Restaurant       41.756192       -88.313986   
4                  Ballydoyle Irish Pub       41.759348       -88.315126   
5                  Jake's Bagels & Deli       41.760742       -88.310206   
6          Taqueria El Tio & Restaurant       41.757515       -88.319157   
7               Two Brothers Roundhouse       41.760639       -88.308788   
8                La Quinta De Los Reyes       41.758603       -88.312294   
9          Holiday Inn Express & Suites       41.759361       -88.309838   
10                          Dairy Queen       41.761321       -88.316895   
11              SciTech Hands On Museum       41.756344       -88.318203   
12                 Riverfront Playhouse       41.757061       -88.313964   
13                         Jimmy John's       41.755421       -88.321381   
14                       RiverEdge Park       41.762860       -88.309188   
15                       Taqueria Muniz       41.751888       -88.328429   
16                             U Samba?       41.759762       -88.315110   
17                             Pancho's       41.754450       -88.300310   
18                     Taqueria Durango       41.754532       -88.314265   
19                       Dollar General       41.760375       -88.322586   
20                 Aurora Pancake House       41.762873       -88.315281   
21  Fifth Third Mortgage - Nancy Alfaro       41.756150       -88.313440   
22                            Walgreens       41.753404       -88.300902   
23              Doughballs Pizza Palace       41.759640       -88.315295   
24                               SUBWAY       41.756954       -88.316280   

                Venue Category  
0                      Theater  
1                          Pub  
2                         Café  
3           Mexican Restaurant  
4                          Pub  
5                   Bagel Shop  
6           Mexican Restaurant  
7                      Brewery  
8           Mexican Restaurant  
9                        Hotel  
10              Ice Cream Shop  
11              Science Museum  
12                     Theater  
13             

In [14]:
# We will find all the unique venues retrieved using Foursquare API
ChicagoSuburbVenues['Venue Category'].unique()

array(['Theater', 'Pub', 'Café', 'Mexican Restaurant', 'Bagel Shop',
       'Brewery', 'Hotel', 'Ice Cream Shop', 'Science Museum',
       'Sandwich Place', 'Park', 'Brazilian Restaurant', 'Discount Store',
       'Breakfast Spot', 'Financial or Legal Service', 'Pharmacy',
       'Pizza Place', 'Gym', 'Fast Food Restaurant', 'Grocery Store',
       'Casino', 'Train Station', 'Chinese Restaurant', 'Clothing Store',
       'Farmers Market', 'Liquor Store', 'Business Service',
       'Food & Drink Shop', 'Gym / Fitness Center', 'Home Service',
       'American Restaurant', 'Shoe Store', 'Tea Room',
       'Seafood Restaurant', 'Bookstore', 'Plaza', 'Coffee Shop',
       'Italian Restaurant', 'Kitchen Supply Store', 'Nail Salon',
       'Trail', 'Bar', 'Frozen Yogurt Shop', 'Dessert Shop',
       'Candy Store', 'Portuguese Restaurant', 'Beer Bar',
       'Japanese Restaurant', 'Snack Place', 'Cosmetics Shop',
       'Noodle House', 'Steakhouse', 'Concert Hall', 'Yoga Studio',
       'Conve

In [15]:
print('There are '+ format(len(ChicagoSuburbVenues['Venue Category'].unique())) + ' unique categories')

There are 227 unique categories


 Using one hot coding we will convert all the uniques venues to columns so that we can filter the data on any venue. 
For our analysis we have to filter the data on "Pizza Place" 

In [16]:
# one hot encoding
suburbs_onehot = pd.get_dummies(ChicagoSuburbVenues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
suburbs_onehot['Suburbs'] = ChicagoSuburbVenues['Suburb'] 

# move neighborhood column to the first column
fixed_columns = [suburbs_onehot.columns[-1]] + list(suburbs_onehot.columns[:-1])
suburbs_onehot = suburbs_onehot[fixed_columns]

suburbs_onehot.head()

Suburbs  ATM  American Restaurant  Arcade  \
0  Aurora, Illinois     0                    0       0   
1  Aurora, Illinois     0                    0       0   
2  Aurora, Illinois     0                    0       0   
3  Aurora, Illinois     0                    0       0   
4  Aurora, Illinois     0                    0       0   

   Argentinian Restaurant  Art Gallery  Arts & Crafts Store  Asian Restaurant  \
0                       0            0                    0                 0   
1                       0            0                    0                 0   
2                       0            0                    0                 0   
3                       0            0                    0                 0   
4                       0            0                    0                 0   

   Assisted Living  Athletics & Sports  ...  Video Store  \
0                0                   0  ...            0   
1                0                   0  ...            0   
2                0                   0  ...            0   
3                0                   0  ...            0   
4                0                   0  ...            0   

   Vietnamese Restaurant  Warehouse Store  Water Park  Weight Loss Center  \
0                      0                0           0                   0   
1                      0                0           0                   0   
2                      0                0           0                   0   
3                      0                0           0                   0   
4                      0                0           0                   0   

   Wine Bar  Winery  Wings Joint  Women's Store  Yoga Studio  
0         0       0            0              0            0  
1         0       0            0              0            0  
2         0       0            0              0            0  
3         0       0            0              0            0  
4         0       0            0              0            0  

[5 rows x 228 columns]

In [17]:
suburbs_onehot.shape

(1377, 228)

In [18]:
suburbs_grouped = suburbs_onehot.groupby('Suburbs').sum().reset_index()
suburbs_grouped

Suburbs  ATM  American Restaurant  Arcade  \
0   Arlington Heights, Illinois     0                    2       0   
1              Aurora, Illinois     0                    0       0   
2              Berwyn, Illinois     0                    0       0   
3         Bolingbrook, Illinois     1                    1       0   
4              Cicero, Illinois     1                    0       0   
5         Des Plaines, Illinois     1                    2       0   
6               Elgin, Illinois     0                    2       0   
7            Evanston, Illinois     0                    2       0   
8                 Gary, Indiana     0                    0       0   
9              Hammond, Indiana     0                    0       0   
10             Joliet, Illinois     0                    1       0   
11           Kenosha, Wisconsin     0                    3       0   
12     Mount Prospect, Illinois     0                    1       0   
13         Naperville, Illinois     0                    3       0   
14            Oak Lawn, Illinois    0                    1       0   
15           Oak Park, Illinois     0                    2       0   
16        Orland Park, Illinois     0                    4       1   
17           Palatine, Illinois     0                    1       0   
18         Schaumburg, Illinois     0                    0       0   
19             Skokie, Illinois     0                    1       0   
20        Tinley Park, Illinois     0                    1       0   
21           Waukegan, Illinois     0                    1       0   
22            Wheaton, Illinois     0                    2       0   

    Argentinian Restaurant  Art Gallery  Arts & Crafts Store  \
0                        0            0                    0   
1                        0            0                    0   
2                        0            0                    1   
3                        0            0                    0   
4                        0            0                    0   
5                        0            0                    0   
6                        0            0                    0   
7                        0            0                    1   
8                        0            0                    0   
9                        0            0                    0   
10                       0            0                    0   
11                       0            1                    0   
12                       0            0                    0   
13                       0            0                    1   
14                       0            0                    0   
15                       0            0                    0   
16                       0            0                    1   
17                       1            0                    0   
18                       0            2                    0   
19                       0            0                    0   
20                       0            0                    0   
21                       0            0                    0   
22                       0            0                    0   

    Asian Restaurant  Assisted Living  Athletics & Sports  ...  Video Store  \
0                  0                0                   0  ...            2   
1                  0                0                   0  ...            0   
2                  0                0                   0  ...            0   
3                  2                0                   0  ...            0   
4                  0                0                   0  ...            0   
5                  0                1                   0  ...            1   
6                  0                0                   0  ...            1   
7                  1                0                   1  ...            2   
8                  0                0                   0  ...            0   
9                

In [19]:
print(suburbs_grouped[['Suburbs','Pizza Place']])

                         Suburbs  Pizza Place
0   Arlington Heights, Illinois             1
1              Aurora, Illinois             3
2              Berwyn, Illinois             4
3         Bolingbrook, Illinois             3
4              Cicero, Illinois             2
5         Des Plaines, Illinois             2
6               Elgin, Illinois             4
7            Evanston, Illinois             5
8                 Gary, Indiana             1
9              Hammond, Indiana             0
10             Joliet, Illinois             2
11           Kenosha, Wisconsin             1
12     Mount Prospect, Illinois             2
13         Naperville, Illinois             5
14            Oak Lawn, Illinois            2
15           Oak Park, Illinois             3
16        Orland Park, Illinois             2
17           Palatine, Illinois             3
18         Schaumburg, Illinois             2
19             Skokie, Illinois             3
20        Tinley Park, Illinois   

In [20]:
df_PizzaPlace=suburbs_grouped[['Suburbs','Pizza Place']]

Now lets Cluster the Pizza Places so that we know the number of pizza places in each City

In [21]:
# set number of clusters
# import k-means from clustering stage
from sklearn.cluster import KMeans
kclusters = 15

suburbs_grouped_clustering = df_PizzaPlace.drop('Suburbs', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(suburbs_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:15] 

/home/jupyterlab/conda/envs/python/lib/python3.6/site-packages/sklearn/cluster/k_means_.py:971: ConvergenceWarning: Number of distinct clusters (6) found smaller than n_clusters (15). Possibly due to duplicate points in X.
  return_n_iter=True)


array([2, 3, 0, 3, 7, 7, 0, 4, 2, 5, 7, 2, 7, 4, 7], dtype=int32)

In [32]:
from sklearn.cluster import KMeans
kmeans = KMeans(n_clusters=3, init='k-means++', max_iter=15, random_state=8)
X = df_PizzaPlace.drop(['Suburbs'], axis=1)

In [33]:
kmeans.fit(X)
kmeans.labels_[0:10]

array([1, 0, 2, 0, 0, 0, 2, 2, 1, 1], dtype=int32)

In [41]:
kclusters=6
suburbs_grouped_clustering = df_PizzaPlace.drop('Suburbs', 1)
#suburbs_grouped_clustering = suburbs_grouped.drop('Suburbs', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(suburbs_grouped_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10] 

array([2, 3, 1, 3, 0, 0, 1, 4, 2, 5], dtype=int32)

In [42]:
#Create a new dataframe that includes the cluster suburb data
df_merged=df_PizzaPlace.copy()
df_merged['Cluster Labels']=kmeans.labels_
df_merged.head(25)

Suburbs  Pizza Place  Cluster Labels
0   Arlington Heights, Illinois             1               2
1              Aurora, Illinois             3               3
2              Berwyn, Illinois             4               1
3         Bolingbrook, Illinois             3               3
4              Cicero, Illinois             2               0
5         Des Plaines, Illinois             2               0
6               Elgin, Illinois             4               1
7            Evanston, Illinois             5               4
8                 Gary, Indiana             1               2
9              Hammond, Indiana             0               5
10             Joliet, Illinois             2               0
11           Kenosha, Wisconsin             1               2
12     Mount Prospect, Illinois             2               0
13         Naperville, Illinois             5               4
14            Oak Lawn, Illinois            2               0
15           Oak Park, Illinois             3               3
16        Orland Park, Illinois             2               0
17           Palatine, Illinois             3               3
18         Schaumburg, Illinois             2               0
19             Skokie, Illinois             3               3
20        Tinley Park, Illinois             4               1
21           Waukegan, Illinois             3               3
22            Wheaton, Illinois             4               1

In [43]:
df_merged = df_merged.join(NearBySuburb.set_index("Suburb"), on="Suburbs")
df_merged

Suburbs  Pizza Place  Cluster Labels   Latitude  \
0   Arlington Heights, Illinois             1               2  42.081156   
1              Aurora, Illinois             3               3  41.757170   
2              Berwyn, Illinois             4               1  41.850587   
3         Bolingbrook, Illinois             3               3  41.700330   
4              Cicero, Illinois             2               0  41.845540   
5         Des Plaines, Illinois             2               0  42.041582   
6               Elgin, Illinois             4               1  42.037260   
7            Evanston, Illinois             5               4  42.044739   
8                 Gary, Indiana             1               2  41.602129   
9              Hammond, Indiana             0               5  41.583366   
10             Joliet, Illinois             2               0  41.526360   
11           Kenosha, Wisconsin             1               2  42.584677   
12     Mount Prospect, Illinois             2               0  42.066417   
13         Naperville, Illinois             5               4  41.772870   
14            Oak Lawn, Illinois            2               0  41.710866   
15           Oak Park, Illinois             3               3  41.887814   
16        Orland Park, Illinois             2               0  41.630663   
17           Palatine, Illinois             3               3  42.110578   
18         Schaumburg, Illinois             2               0  42.033361   
19             Skokie, Illinois             3               3  42.033369   
20        Tinley Park, Illinois             4               1  41.573367   
21           Waukegan, Illinois             3               3  42.363633   
22            Wheaton, Illinois             4               1  41.864696   

    Longitude  
0  -87.980216  
1  -88.314754  
2  -87.793668  
3  -88.071771  
4  -87.754020  
5  -87.887392  
6  -88.281099  
7  -87.693046  
8  -87.337137  
9  -87.500043  
10 -88.084021  
11 -87.821226  
12 -87.937291  
13 -88.147928  
14 -87.758108  
15 -87.788762  
16 -87.853629  
17 -88.033689  
18 -88.083406  
19 -87.733397  
20 -87.784494  
21 -87.844794  
22 -88.110171

In [44]:
df_merged.sort_values(['Cluster Labels'],inplace=True)

In [45]:
df_merged

Suburbs  Pizza Place  Cluster Labels   Latitude  \
18         Schaumburg, Illinois             2               0  42.033361   
4              Cicero, Illinois             2               0  41.845540   
5         Des Plaines, Illinois             2               0  42.041582   
16        Orland Park, Illinois             2               0  41.630663   
14            Oak Lawn, Illinois            2               0  41.710866   
12     Mount Prospect, Illinois             2               0  42.066417   
10             Joliet, Illinois             2               0  41.526360   
22            Wheaton, Illinois             4               1  41.864696   
20        Tinley Park, Illinois             4               1  41.573367   
2              Berwyn, Illinois             4               1  41.850587   
6               Elgin, Illinois             4               1  42.037260   
0   Arlington Heights, Illinois             1               2  42.081156   
8                 Gary, Indiana             1               2  41.602129   
11           Kenosha, Wisconsin             1               2  42.584677   
15           Oak Park, Illinois             3               3  41.887814   
17           Palatine, Illinois             3               3  42.110578   
3         Bolingbrook, Illinois             3               3  41.700330   
19             Skokie, Illinois             3               3  42.033369   
1              Aurora, Illinois             3               3  41.757170   
21           Waukegan, Illinois             3               3  42.363633   
13         Naperville, Illinois             5               4  41.772870   
7            Evanston, Illinois             5               4  42.044739   
9              Hammond, Indiana             0               5  41.583366   

    Longitude  
18 -88.083406  
4  -87.754020  
5  -87.887392  
16 -87.853629  
14 -87.758108  
12 -87.937291  
10 -88.084021  
22 -88.110171  
20 -87.784494  
2  -87.793668  
6  -88.281099  
0  -87.980216  
8  -87.337137  
11 -87.821226  
15 -87.788762  
17 -88.033689  
3  -88.071771  
19 -87.733397  
1  -88.314754  
21 -87.844794  
13 -88.147928  
7  -87.693046  
9  -87.500043

Now lets plot the clusters

In [46]:
latitude= 41.8755616
longtitude=-87.6244212

map_clusters = folium.Map(location=[latitude, longtitude], zoom_start=10)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i + x + (i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(df_merged['Latitude'], df_merged['Longitude'], df_merged['Suburbs'], df_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(map_clusters)
       
map_clusters

The area marked in orange has zero places

In [47]:
# Analysing ech cluster
df_merged.loc[df_merged['Cluster Labels'] == 5, df_merged.columns[[0] + list(range(1, df_merged.shape[1]))]]

Suburbs  Pizza Place  Cluster Labels   Latitude  Longitude
9  Hammond, Indiana             0               5  41.583366 -87.500043

In [48]:
df_merged.loc[df_merged['Cluster Labels'] == 2, df_merged.columns[[0] + list(range(1, df_merged.shape[1]))]]

Suburbs  Pizza Place  Cluster Labels   Latitude  \
0   Arlington Heights, Illinois             1               2  42.081156   
8                 Gary, Indiana             1               2  41.602129   
11           Kenosha, Wisconsin             1               2  42.584677   

    Longitude  
0  -87.980216  
8  -87.337137  
11 -87.821226

In [49]:
suburb_population_format

suburb population
0              Aurora, Illinois      198870
1          Naperville, Illinois      149196
2              Joliet, Illinois      148227
3               Elgin, Illinois      111401
4            Kenosha, Wisconsin      101124
5            Waukegan, Illinois       85720
6              Cicero, Illinois       79943
7         Bolingbrook, Illinois       76468
8   Arlington Heights, Illinois       74593
9              Hammond, Indiana       74423
10                Gary, Indiana       74879
11           Evanston, Illinois       72836
12         Schaumburg, Illinois       72387
13           Palatine, Illinois       67143
14             Skokie, Illinois       62088
15        Des Plaines, Illinois       58243
16        Orland Park, Illinois       57634
17        Tinley Park, Illinois       55446
18             Berwyn, Illinois       53803
19     Mount Prospect, Illinois       53218
20            Wheaton, Illinois       52914
21           Oak Park, Illinois       52443

In [50]:
ChicagoSuburbVenues.loc[(ChicagoSuburbVenues['Suburb'] == 'Arlington Heights, Illinois ')]
#ChicagoSuburbVenues

Suburb  Suburb Latitude  Suburb Longitude  \
480  Arlington Heights, Illinois         42.081156        -87.980216   
481  Arlington Heights, Illinois         42.081156        -87.980216   
482  Arlington Heights, Illinois         42.081156        -87.980216   
483  Arlington Heights, Illinois         42.081156        -87.980216   
484  Arlington Heights, Illinois         42.081156        -87.980216   
..                            ...              ...               ...   
545  Arlington Heights, Illinois         42.081156        -87.980216   
546  Arlington Heights, Illinois         42.081156        -87.980216   
547  Arlington Heights, Illinois         42.081156        -87.980216   
548  Arlington Heights, Illinois         42.081156        -87.980216   
549  Arlington Heights, Illinois         42.081156        -87.980216   

                                  Venue  Venue Latitude  Venue Longitude  \
480                             altThai       42.082113       -87.980847   
481                           Starbucks       42.082282       -87.981517   
482        Sweet T's Bakery-Cake Studio       42.081974       -87.982570   
483   Metropolis Performing Arts Center       42.082871       -87.984665   
484            Bentley's Corner Barkery       42.081130       -87.981808   
..                                  ...             ...              ...   
545       Pioneer Park Community Center       42.076141       -87.992929   
546  Metra Union Pacific Northwest# 652       42.076601       -87.966393   
547                    Metra UP-NW Line       42.088608       -87.993289   
548     Arlington Heights Train Station       42.078275       -87.963649   
549              Standard Pipe & Supply       42.073130       -87.966452   

                 Venue Category  
480             Thai Restaurant  
481                 Coffee Shop  
482                      Bakery  
483                     Theater  
484                   Pet Store  
..                          ...  
545                         Gym  
546          Light Rail Station  
547                       Train  
548               Train Station  
549  Construction & Landscaping  

[70 rows x 7 columns]

## Results

The results of the analysis are,
Cluster 4 has most number of pizza places which is five where as Cluster 5 has Zero pizza place.
The only place in Cluster 5 is Hammond, Indiana.Since our objective is to start a Pizza place in Illinois near Chicago , lets look at the the cluster that has atleast one pizza place. Below are the list of places that has atleast one pizza place.
##### Arlington Heights, Illinois
##### Gary, Indiana
##### Kenosha, Wisconsin
Arlington Heights, Illinois  has an estimated population of 74,593 and has many other attractions like theaters, train stations,stores etc. So an ideal location to start a new Pizza place will be Arlington Heights, Illinois.